adapters/caaqm.js

```diff
@@ -29,131 +32,88 @@ export function fetchData (source, cb) {
+              coords: {latitude: Number(site['latitude']), longitude: Number(site['longitude'])}
+            });
+          }
+        }
+      },
+      new DataStream()
+    )
+    .into(
+      async (measurements, {coords, station_id: stationId}) => {
         const options = Object.assign(requestOptions, {
           url: 'https://app.cpcbccr.com/caaqms/caaqms_view_data',
-          body: Buffer.from(`{"site_id":"${s['station_id']}"}`).toString('base64')
-        });
-
-        return request(options, (err, res, body) => {
-          if (err || !res.statusCode === 200) {
-            return done(null, {});
-          }
-
-          body = safeParse(body);
-          if (body === undefined) {
-            return done(null, {});
-          }
-
-          return done(null, body);
+          body: Buffer.from(`{"site_id":"${stationId}"}`).toString('base64'),
+          resolveWithFullResponse: true
         });
-      };
-    });
 
-    // Run the async requests
-    parallelLimit(tasks, 5, (err, results) => {
-      if (err) {
-        return cb({message: 'Failure to load data urls.'});
-      }
-
-      // Wrap everything in a try/catch in case something goes wrong
-      try {
-        // Format the data
-        const data = formatData({sites: sites, results: results});
-        // Make sure the data is valid
-        if (data === undefined) {
-          return cb({message: 'Failure to parse data.'});
+        try {
+          const response = await rp(options);
+
+          const {siteInfo, tableData: {bodyContent}} = JSON.parse(response.body);
+
+          await (
+            DataStream
+              .from(bodyContent)
+              .each(async p => {
+                let parameter = p.parameters.toLowerCase().replace('.', '');
+                parameter = (parameter === 'ozone') ? 'o3' : parameter;
+
+                // Make sure we want the pollutant
+                if (!acceptableParameters.includes(parameter)) {
+                  return;
+                }
+
+                let m = {
+                  averagingPeriod: {unit: 'hours', value: 0.25},
+                  city: siteInfo.city,
+                  location: siteInfo.siteName,
+                  coordinates: coords,
+                  attribution: [{
+                    name: 'Central Pollution Control Board',
+                    url: 'https://app.cpcbccr.com/ccr/#/caaqm-dashboard-all/caaqm-landing'
+                  }],
+                  parameter: parameter,
+                  unit: p.unit,
+                  value: Number(p.concentration)
+                };
+
+                // Date
+                const date = moment.tz(`${p.date} ${p.time}`, 'DD MMM YYYY HH:mm', 'Asia/Kolkata');
+                m.date = {utc: date.toDate(), local: date.format()};
+
+                await measurements.whenWrote(m);
+              })
+              .run()
+          );
+        } catch (e) {
+          throw new FetchError(`Cannot fetch data for station id`, source, e);
```

Comment:
I think the `e` here is maybe catching HTML? Can we just hardcode with something descriptive? I'm seeing output like below which I'd rather not have saved/output in case we end up with large HTML error pages.

```
2018-10-17T12:10:08.689Z - verbose: fetch data for station i: 404 - "<html>\r\n<head><title>404 Not Found</title></head>\r\n<body bgcolor=\"white\">\r\n<center><h1>404 Not Found</h1></center>\r\n<hr><center>nginx</center>\r\n</body>\r\n</html>\r\n"
2018-10-17T12:10:08.791Z - verbose: fetch data for station i: 404 - "<html>\r\n<head><title>404 Not Found</title></head>\r\n<body bgcolor=\"white\">\r\n<center><h1>404 Not Found</h1></center>\r\n<hr><center>nginx</center>\r\n</body>\r\n</html>\r\n"
```

```
2018-10-17T12:10:30.186Z - info:  itemsInserted=163, timeStarted=1539778183098, results=[message=[Dry Run] New measurements found for caaqm: 163, fetch data for station i: 404 - "<html>\r\n<head><title>404 Not Found</title></head>\r\n<body bgcolor=\"white\">\r\n<center><h1>404 Not Found</h1></center>\r\n<hr><center>nginx</center>\r\n</body>\r\n</html>\r\n"=39, count=163, duration=46.705, sourceName=caaqm], fetch data for station i: 404 - "<html>\r\n<head><title>404 Not Found</title></head>\r\n<body bgcolor=\"white\">\r\n<center><h1>404 Not Found</h1></center>\r\n<hr><center>nginx</center>\r\n</body>\r\n</html>\r\n"=39, timeEnded=1539778230186
```

Reply:
There may be some kind of toString() implemented. I'll see what I can dig up - the 404 part would be good to have in the message.

Reply:
@jflasher - there's a single message for this now. I changed the error display a little bit so now the message will be taken into account. It won't affect other errors display, as this is the single one place this is used for now.